public/app/plugins/datasource/prometheus/types.ts

```diff
@@ -165,3 +166,21 @@ export enum LegendFormatMode {
   Verbose = '__verbose',
   Custom = '__custom',
 }
+
+export enum PromVariableQueryType {
+  LabelNames,
+  LabelValues,
+  MetricNames,
+  VarQueryResult,
+  SeriesQuery,
+}
+
+export interface PromVariableQuery extends DataQuery {
+  query?: string;
+  expr?: string;
+  exprType?: PromVariableQueryType;
+  label?: string;
+  metric?: string;
+  varQuery?: string;
+  seriesQuery?: string;
```

Comment:
Should all of those be optional? For instance, I think we always need the `exprType`.

Reply:
That's true, exprType is required. The other fields are only necessary based on the exprType chosen